File: test_data/golang/test_main/aas_core_meta.v3/input/snippets/Verification/extension_names_are_unique.go
// Check that all [aastypes.IExtension.Name] are unique among
// `extensions`.
func ExtensionNamesAreUnique[E aastypes.IExtension](
	extensions []E) bool {
	nameSet := make(map[string]struct{})

	for _, extension := range extensions {
		name := extension.Name()
		_, has := nameSet[name]
		if has {
			return false
		}

		nameSet[name] = struct{}{}
	}
	return true
}

File: test_data/java/test_main/aas_core_meta.v3/input/snippets/Verification/extension_names_are_unique.java
/**
* Check that all {@link IExtension#getName() name}  are unique among extensions.
* @param extensions the extensions
*/
public static boolean extensionNamesAreUnique(
  Iterable<? extends IExtension> extensions) {
  Objects.requireNonNull(extensions);

  Set<String> nameSet = new HashSet<>();

  for (IExtension extension : extensions) {
    if (nameSet.contains(extension.getName())) {
      return false;
    }
    nameSet.add(extension.getName());
  }
  return true;
}
